Sources/Abstract/Types/Algebraic/Inclusive.swift

```diff
@@ -0,0 +1,97 @@
+#if SWIFT_PACKAGE
+    import Operadics
+#endif
+
+/// Inclusive is the "inclusive-or" type: represents either something, or somehting else, or both.
+
+// sourcery: fixedTypesForPropertyBasedTests = "TestStructure,TestStructure"
+public enum Inclusive<A,B> {
```

Comment:
I'm familiar with Haskell's `These` and I think the name is ugly and uninformative (a nice wordplay, though). `Inclusive` literally represents the logical inclusive-or, so I think it's only natural to call it like that.

I used `Product` and `Coproduct` (rather that `Pair` and `Either`) because I prefer a category-theoretical naming convention, and to leverage dualism, when possible. Haskell is also not a great source for the basic "sum" type, because `Either` in Haskell, while having a bifunctor definition, is typically used to represent the possibility of failure, so it's a right-biased functor, following the convention that *right* has two meanings, something that is much more clearly expressed by a `Result` type with `.success` and `.failure` cases (like in Rust, and conventionally in Swift).

Thus, I see no reason to call the root sum type `Either`: I could have called it `Exclusive`, but as I said I prefer to highlight the dualism with `Product`.